linkerd/meshtls/boring/src/creds/verify.rs

```diff
@@ -0,0 +1,78 @@
+use boring::x509::X509;
+use linkerd_dns_name as dns;
+use linkerd_identity as id;
+use std::io;
+
+pub(crate) fn verify_id(cert: &X509, id: &id::Id) -> io::Result<()> {
+    for san in cert.subject_alt_names().into_iter().flatten() {
+        if let Some(n) = san.dnsname() {
+            if let Ok(name) = n.parse::<dns::Name>() {
+                if *name == id.to_str() {
+                    return Ok(());
+                }
+            }
+            tracing::warn!("DNS SAN name {} could not be parsed", n)
+        }
+    }
+
+    Err(io::Error::new(
+        io::ErrorKind::Other,
+        "certificate does not match TLS identity",
+    ))
+}
+
+#[cfg(test)]
+mod tests {
+    use super::verify_id;
+    use boring::x509::X509;
+    use linkerd_identity as id;
+    use rcgen::generate_simple_self_signed;
+    use std::str::FromStr;
+
+    fn generate_cert_with_name(name: Option<&str>) -> X509 {
+        let sans = name.map(|s| vec![s.into()]).unwrap_or_default();
```

Comment:
it looks like we don't have any tests for certificates with multiple SANs --- we only ever generate certs with 0-1 SANs. since we have logic for iterating over multiple matching and non-matching SANs, we should probably have tests for certs with more than one SAN...